File: Cocos-Creator/BunnyMark/assets/Script/Moves.js
cc.Class({
    extends: cc.Component,
    
    // variables of this class
    properties: {
        direction: 1
    },

    // use this for initialization
    onLoad: function () {
        // setting random start position of the sprite
        this.node.x = Math.floor((Math.random() * 740 ) + 30);
        this.node.y = Math.floor((Math.random() * 560 ) + 10);
    },

    // called every frame, uncomment this function to activate update callback
    update: function (dt) {
        // moving this sprite
        this.node.y -= 5*this.direction;
        // if it goes out of the screen
        if(this.node.y < 20 ){
            this.direction = -1;
            this.node.y = 20;
        }
        if(this.node.y > 580 ){
            this.direction = 1;
            this.node.y = 580;
        }
    },
});

File: Unity 5/BunnyMark/Assets/Scripts/Moves.cs
using UnityEngine;
using System.Collections;

public class Moves : MonoBehaviour {

	// constant size: half height of the board
	// it's constant cause it's set to Fit Height for multiresolutions
	const float size = 4.85f;
	// direction of moving: 1 = down, -1 = up
	int direction = 1;
	// screen size values
	float height,width,ratio;

	// Use this for initialization
	void Start(){
		// get Screen height and width
		height = Screen.height;
		width = Screen.width;
		// count ratio of width and height
		ratio = width / height;
		// multiply ratio by height of the board
		// width range: <-ratio,ratio>
		ratio = ratio * size;
		// set random position of the prefab width in range <-ratio,ratio>
		this.transform.position = new Vector2(Random.Range(-ratio,ratio),Random.Range(-size,size));
	}

	// Update is called once per frame
	void Update () {
		// update position of this rabbit
		this.transform.position = new Vector2 (this.transform.position.x, this.transform.position.y - 0.07f * direction);
		// if rabbit is going out of the screen, change direction of moving
		if (this.transform.position.y < -size) {
			direction = -1;
			this.transform.position = new Vector2(this.transform.position.x, -size);
		}
		// if rabbit is going out of the screen, change direction of moving
		if (this.transform.position.y > size) {
			direction = 1;
			this.transform.position = new Vector2(this.transform.position.x, size);
		}
	}
}
